src/js/player.js

```diff
@@ -2062,6 +2070,26 @@ class Player extends Component {
   }
 
   /**
+   * Returns the current source objects.
+   *
+   * @return {Object[]} The current source objects
+   * @method currentSources
+   */
+  currentSources() {
+    return this.cache_.sources || [this.currentSource()];
+  }
+
+  /**
+   * Returns the current source object.
+   *
+   * @return {Object} The current source object
+   * @method currentSource
+   */
+  currentSource() {
+    return this.cache_.source || { src: this.currentSrc() };
```

Comment:
Having `currentSource` return `undefined` seems inconsistent to me. Both from it's internal type signature but also between the 3 methods.
`currentSrc` always returns a string and `currentSources` always returns an array, why should `currentSource` be different?